andela/ah-scorpion-frontend
# #159993335 Allow for user registration

#### What does this PR do?

- Allows for user signUp.

#### Description of Task to be completed?
- Handle disabling the Signup button to indicate loading status during SignUp.
- Handle successful user signUp and display the message asking the user to confirm their email address before proceeding.
- Handle unsuccessful signUp and display descriptive error messages for the errors raised.

#### How should this be manually tested?

- Clone this branch `ft-user-registration-159993335`.
- Go to the signup endpoint

- _When you try signing up a user:_

  - You should see the SignUp button get disabled to indicate that the request is pending.
  - You should get an alert box with the message `Please confirm your email address to complete the registration` in case of a successful request.
  - You should get an alert box with the relevant error message in case of an invalid request.

#### Any background context you want to provide?

- None

#### What are the relevant pivotal tracker stories?
[#159993335](https://www.pivotaltracker.com/story/show/159993335)
